lib/plugin/instance.rb

```diff
@@ -82,6 +82,12 @@ def initialize(metadata = nil, path = nil)
     @idx = 0
   end
 
+  def register_anonymous_cache_key(key, &block)
+    key_method = "key_#{key}"
+    add_to_class(Middleware::AnonymousCache, key_method, &block)
+    Middleware::AnonymousCache.cache_key_segments[key] = key_method
```

Comment:
overall the change looks good to me but I would re-run `compile_key_builder` here to ensure that if for any reason it was ran earlier it will be re-cached.

Reply:
Sure, good idea!